providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/RemoveQueueButton.tsx

```diff
@@ -0,0 +1,156 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  Button,
+  CloseButton,
+  Dialog,
+  IconButton,
+  Portal,
+  Text,
+  useDisclosure,
+  VStack,
+  For,
+} from "@chakra-ui/react";
+import { useUiServiceRemoveWorkerQueue } from "openapi/queries";
+import type { Worker } from "openapi/requests/types.gen";
+import { useState } from "react";
+import { LuListMinus } from "react-icons/lu";
+
+interface RemoveQueueButtonProps {
+  onQueueUpdate: (toast: Record<string, string>) => void;
+  worker: Worker;
+}
+
+export const RemoveQueueButton = ({ onQueueUpdate, worker }: RemoveQueueButtonProps) => {
+  const { onClose, onOpen, open } = useDisclosure();
+  const [selectedQueue, setSelectedQueue] = useState<string>("");
+
+  const removeQueueMutation = useUiServiceRemoveWorkerQueue({
+    onError: (error) => {
+      onQueueUpdate({
+        description: `Unable to remove queue from worker ${worker.worker_name}: ${error}`,
+        title: "Remove Queue Failed",
+        type: "error",
+      });
+    },
+    onSuccess: () => {
+      onQueueUpdate({
+        description: `Queue "${selectedQueue}" was removed from worker ${worker.worker_name}.`,
+        title: "Queue Removed",
+        type: "success",
+      });
+      onClose();
+      setSelectedQueue("");
+    },
+  });
+
+  const handleRemoveQueue = () => {
+    if (!selectedQueue) {
+      onQueueUpdate({
+        description: "Please select a queue to remove.",
+        title: "Invalid Selection",
+        type: "error",
+      });
+      return;
+    }
+
+    removeQueueMutation.mutate({
+      requestBody: { queue_name: selectedQueue },
+      workerName: worker.worker_name,
+    });
+  };
+
+  const availableQueues = worker.queues || [];
+  const isDisabled = availableQueues.length === 0;
+
+  return (
+    <>
+      <IconButton
+        size="sm"
+        variant="ghost"
+        onClick={onOpen}
+        aria-label="Remove Queue"
+        title="Remove Queue"
+        colorPalette="danger"
+        disabled={isDisabled}
+      >
```

Comment:
Can you just "hide" the button (return `null`) instead of disabling? e.g. like:
```suggestion
  const isDisabled = availableQueues.length === 0;
  if (isDisabled) {
    return null;
  }

  return (
    <>
      <IconButton
        size="sm"
        variant="ghost"
        onClick={onOpen}
        aria-label="Remove Queue"
        title="Remove Queue"
        colorPalette="danger"
      >
```

Reply:
Done! 

<img width="1916" height="652" alt="image" src="https://github.com/user-attachments/assets/ba59ca19-ec88-4105-8db9-92ed92f409ee" />